### PR TITLE
Jbeap 10146 branch 3.0

### PR DESCRIPTION
**Pull request by rsearls**

### CODE REVIEW

**Comment by rsearls** [comment]
files revised

File: testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java [hunk 1]
```diff
@@ -77,6 +82,16 @@ private static WebArchive createDeploymentWithSpringMvcDispatcher() {
...
                 .addAsWebInfResource(AddSpringResteasyAsResourceRootTest.class.getPackage(), "mvc-dispatcher-servlet/applicationContext.xml", "applicationContext.xml");
+
+        archive.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+            new PropertyPermission("arquillian.*", "read"),
+            new ReflectPermission("suppressAccessChecks"),
+            new RuntimePermission("accessDeclaredMembers"),
+            new RuntimePermission("getClassLoader"),
+            new FilePermission("<<ALL FILES>>", "read"),
+            new LoggingPermission("control", "")
+        ), "permissions.xml");
```
Comment (kanovotn) [inline testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java]: This note it is still valid. At least the test doesn't pass with the specified permissions.

File: testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java [hunk 2]
```diff
@@ -103,6 +118,16 @@ public void testDeploymentWithSpringContextLoaderListenerAddsResourceRoot() thro
...
                 .addAsWebInfResource(AddSpringResteasyAsResourceRootTest.class.getPackage(), "applicationContext.xml", "applicationContext.xml");
+
+        archive.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+            new PropertyPermission("arquillian.*", "read"),
+            new ReflectPermission("suppressAccessChecks"),
+            new RuntimePermission("accessDeclaredMembers"),
+            new RuntimePermission("getClassLoader"),
+            new FilePermission("<<ALL FILES>>", "read"),
+            new LoggingPermission("control", "")
+        ), "permissions.xml");
```
Comment (kanovotn) [inline testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java]: Still valid.

File: testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java [hunk 3]
```diff
@@ -130,6 +155,16 @@ public void testDeploymentWithoutSpringMvcDispatcherOrListenerDoesNotAddResource
         WebArchive archive = TestUtil.prepareArchive(deploymentWithoutSpringMvcDispatcherOrListenerSpringIncluded);
         archive.addAsWebInfResource(AddSpringResteasyAsResourceRootTest.class.getPackage(), "web-no-mvc-no-listener.xml", "web.xml")
                 .addAsWebInfResource(AddSpringResteasyAsResourceRootTest.class.getPackage(), "applicationContext.xml", "applicationContext.xml");
+
+        archive.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
```
Comment (kanovotn) [inline testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java]: Still valid.

File: testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java (hunk 1 above)
Comment (rsearls) [inline testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java]: I can not reproduce the issue of needing to add permission "insertProvider".
This permission block has a comment, "remaining permissions needed to run springframework"

File: testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java (hunk 2 above)
Comment (rsearls) [inline testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java]: I can not reproduce this.  All the listed permission are required.
I added a comment to the permission block.

File: testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/ContextRefreshDependenciesInDeploymentTest.java [hunk 4]
```diff
@@ -40,6 +46,16 @@
                 .addClass(NotForWildFly9.class) //required as this test is not @RunAsClient annotated
                 .addAsWebInfResource(ContextRefreshDependenciesInDeploymentTest.class.getPackage(), "web.xml", "web.xml")
                 .addAsWebInfResource(ContextRefreshDependenciesInDeploymentTest.class.getPackage(), "contextRefresh/applicationContext.xml", "applicationContext.xml");
+
+        archive.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+            new PropertyPermission("arquillian.*", "read"),
+            new ReflectPermission("suppressAccessChecks"),
+            new RuntimePermission("accessDeclaredMembers"),
+            new RuntimePermission("getClassLoader"),
```
Comment (rsearls) [inline testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/ContextRefreshDependenciesInDeploymentTest.java]: I could not reproduce this.  I needed all the permissions listed.

File: testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java (hunk 1 above)
Comment (kanovotn) [inline testsuite/integration-tests-spring/deployment/src/test/java/org/jboss/resteasy/test/spring/deployment/AddSpringResteasyAsResourceRootTest.java]: I get the failure intermittently. It is reproducible when I comment out deployment dep1 and dep2 and run the AddSpringResteasyAsResourceRootTest. First time it fails, subsequent runs pass.

Independently on this intermittent issue, i see these info messages from spring in the log:

> 17:40:30,618 INFO  [org.springframework.web.context.support.StandardServletEnvironment] (ServerService Thread Pool -- 19) Caught AccessControlException when accessing system property [spring.liveBeansView.mbeanDomain]; its value will be returned [null]. Reason: WFSM000001: Permission check failed (permission "("java.util.PropertyPermission" "spring.liveBeansView.mbeanDomain" "read")" in code source "(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)" of "null")

But the test passes even with them.

**Comment by rsearls** [comment]
Katerina,

I've run AddSpringResteasyAsResourceRootTest with permissions suppressAccessChecks and accessDeclaredMembers commented out.
Below is the stacktrace if the permission failure.  Review the stacktrace.  You will see that there
is no resteasy class listed.  Hence there is no place for me to attempt to perform the needed security check,
this is why these permissions are required as is.

12:47:30,937 ERROR [org.springframework.web.servlet.DispatcherServlet] (ServerService Thread Pool -- 26) Context initialization failed: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "accessDeclaredMembers")" in code source "(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)" of "ModuleClassLoader for Module "deployment.deploymentWithSpringMvcDispatcherSpringIncluded.war" from Service Module Loader")
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:278)
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:175)
	at java.lang.Class.checkMemberAccess(Class.java:2348)
	at java.lang.Class.getDeclaredMethods(Class.java:1974)
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:609)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:521)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:507)
	at org.springframework.util.ReflectionUtils.getUniqueDeclaredMethods(ReflectionUtils.java:567)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.getTypeForFactoryMethod(AbstractAutowireCapableBeanFactory.java:683)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineTargetType(AbstractAutowireCapableBeanFactory.java:627)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.predictBeanType(AbstractAutowireCapableBeanFactory.java:597)
	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1445)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doGetBeanNamesForType(DefaultListableBeanFactory.java:445)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanNamesForType(DefaultListableBeanFactory.java:415)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:86)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:678)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:520)
	at org.springframework.web.servlet.FrameworkServlet.configureAndRefreshWebApplicationContext(FrameworkServlet.java:668)
	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:634)
	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:682)
	at org.springframework.web.servlet.FrameworkServlet.initWebApplicationContext(FrameworkServlet.java:553)
	at org.springframework.web.servlet.FrameworkServlet.initServletBean(FrameworkServlet.java:494)
	at org.springframework.web.servlet.HttpServletBean.init(HttpServletBean.java:136)
	at javax.servlet.GenericServlet.init(GenericServlet.java:244)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:117)
	at org.wildfly.extension.undertow.security.RunAsLifecycleInterceptor.init(RunAsLifecycleInterceptor.java:78)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:103)
	at io.undertow.servlet.core.ManagedServlet$DefaultInstanceStrategy.start(ManagedServlet.java:250)
	at io.undertow.servlet.core.ManagedServlet.createServlet(ManagedServlet.java:133)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:558)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:529)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at io.undertow.servlet.core.DeploymentManagerImpl.start(DeploymentManagerImpl.java:571)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:100)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jboss.threads.JBossThread.run(JBossThread.java:320)

12:47:30,939 ERROR [org.jboss.msc.service.fail] (ServerService Thread Pool -- 26) MSC000001: Failed to start service jboss.undertow.deployment.default-server.default-host./deploymentWithSpringMvcDispatcherSpringIncluded: org.jboss.msc.service.StartException in service jboss.undertow.deployment.default-server.default-host./deploymentWithSpringMvcDispatcherSpringIncluded: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "accessDeclaredMembers")" in code source "(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)" of "ModuleClassLoader for Module "deployment.deploymentWithSpringMvcDispatcherSpringIncluded.war" from Service Module Loader")
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:84)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
	at org.jboss.threads.JBossThread.run(JBossThread.java:320)
Caused by: java.security.AccessControlException: WFSM000001: Permission check failed (permission "("java.lang.RuntimePermission" "accessDeclaredMembers")" in code source "(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)" of "ModuleClassLoader for Module "deployment.deploymentWithSpringMvcDispatcherSpringIncluded.war" from Service Module Loader")
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:278)
	at org.wildfly.security.manager.WildFlySecurityManager.checkPermission(WildFlySecurityManager.java:175)
	at java.lang.Class.checkMemberAccess(Class.java:2348)
	at java.lang.Class.getDeclaredMethods(Class.java:1974)
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:609)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:521)
	at org.springframework.util.ReflectionUtils.doWithMethods(ReflectionUtils.java:507)
	at org.springframework.util.ReflectionUtils.getUniqueDeclaredMethods(ReflectionUtils.java:567)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.getTypeForFactoryMethod(AbstractAutowireCapableBeanFactory.java:683)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineTargetType(AbstractAutowireCapableBeanFactory.java:627)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.predictBeanType(AbstractAutowireCapableBeanFactory.java:597)
	at org.springframework.beans.factory.support.AbstractBeanFactory.isFactoryBean(AbstractBeanFactory.java:1445)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doGetBeanNamesForType(DefaultListableBeanFactory.java:445)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.getBeanNamesForType(DefaultListableBeanFactory.java:415)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:86)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:678)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:520)
	at org.springframework.web.servlet.FrameworkServlet.configureAndRefreshWebApplicationContext(FrameworkServlet.java:668)
	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:634)
	at org.springframework.web.servlet.FrameworkServlet.createWebApplicationContext(FrameworkServlet.java:682)
	at org.springframework.web.servlet.FrameworkServlet.initWebApplicationContext(FrameworkServlet.java:553)
	at org.springframework.web.servlet.FrameworkServlet.initServletBean(FrameworkServlet.java:494)
	at org.springframework.web.servlet.HttpServletBean.init(HttpServletBean.java:136)
	at javax.servlet.GenericServlet.init(GenericServlet.java:244)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:117)
	at org.wildfly.extension.undertow.security.RunAsLifecycleInterceptor.init(RunAsLifecycleInterceptor.java:78)
	at io.undertow.servlet.core.LifecyleInterceptorInvocation.proceed(LifecyleInterceptorInvocation.java:103)
	at io.undertow.servlet.core.ManagedServlet$DefaultInstanceStrategy.start(ManagedServlet.java:250)
	at io.undertow.servlet.core.ManagedServlet.createServlet(ManagedServlet.java:133)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:558)
	at io.undertow.servlet.core.DeploymentManagerImpl$2.call(DeploymentManagerImpl.java:529)
	at io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:42)
	at io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1505)
	at io.undertow.servlet.core.DeploymentManagerImpl.start(DeploymentManagerImpl.java:571)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService.startContext(UndertowDeploymentService.java:100)
	at org.wildfly.extension.undertow.deployment.UndertowDeploymentService$1.run(UndertowDeploymentService.java:81)
	... 6 more

12:47:30,944 ERROR [org.jboss.as.controller.management-operation] (management-handler-thread - 2) WFLYCTL0013: Operation ("deploy") failed - address: ([("deployment" => "deploymentWithSpringMvcDispatcherSpringIncluded.war")]) - failure description: {"WFLYCTL0080: Failed services" => {"jboss.undertow.deployment.default-server.default-host./deploymentWithSpringMvcDispatcherSpringIncluded" => "java.security.AccessControlException: WFSM000001: Permission check failed (permission \"(\"java.lang.RuntimePermission\" \"accessDeclaredMembers\")\" in code source \"(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)\" of \"ModuleClassLoader for Module \"deployment.deploymentWithSpringMvcDispatcherSpringIncluded.war\" from Service Module Loader\")
    Caused by: java.security.AccessControlException: WFSM000001: Permission check failed (permission \"(\"java.lang.RuntimePermission\" \"accessDeclaredMembers\")\" in code source \"(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)\" of \"ModuleClassLoader for Module \"deployment.deploymentWithSpringMvcDispatcherSpringIncluded.war\" from Service Module Loader\")"}}
12:47:30,947 ERROR [org.jboss.as.server] (management-handler-thread - 2) WFLYSRV0021: Deploy of deployment "deploymentWithSpringMvcDispatcherSpringIncluded.war" was rolled back with the following failure message: 
{"WFLYCTL0080: Failed services" => {"jboss.undertow.deployment.default-server.default-host./deploymentWithSpringMvcDispatcherSpringIncluded" => "java.security.AccessControlException: WFSM000001: Permission check failed (permission \"(\"java.lang.RuntimePermission\" \"accessDeclaredMembers\")\" in code source \"(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)\" of \"ModuleClassLoader for Module \"deployment.deploymentWithSpringMvcDispatcherSpringIncluded.war\" from Service Module Loader\")
    Caused by: java.security.AccessControlException: WFSM000001: Permission check failed (permission \"(\"java.lang.RuntimePermission\" \"accessDeclaredMembers\")\" in code source \"(vfs:/content/deploymentWithSpringMvcDispatcherSpringIncluded.war/WEB-INF/lib/spring-core-4.2.4.RELEASE.jar <no signer certificates>)\" of \"ModuleClassLoader for Module \"deployment.deploymentWithSpringMvcDispatcherSpringIncluded.war\" from Service Module Loader\")"}}





----- Original Message -----

**Comment by kanovotn** [comment]
@rsearls Thank you for explanation. I don't have more comments. LGTM.

**Pull request merged**